src/mappings/clearingHouse.ts

```diff
@@ -391,45 +391,51 @@ export function handleFundingPaymentSettled(event: FundingPaymentSettledEvent):
-    let positionChangedLogIndex = event.logIndex.minus(BigInt.fromI32(1))
+    const positionChangedLogIndex = event.logIndex.minus(BigInt.fromI32(1))
     // // the referral event shares the same tx as the positionChanged event
-    let positionChangedEvent = PositionChanged.load(
+    const positionChangedEvent = PositionChanged.load(
         event.transaction.hash.toHexString() + "-" + positionChangedLogIndex.toString(),
     )
-    let code = event.params.referralCode.toString()
-
-    if (positionChangedEvent !== null) {
-        let trader = getOrCreateTrader(event.transaction.from)
-        let referralCode = getReferralCode(code)
-        if (referralCode !== null) {
-            let referralCodeDayData = getReferralCodeDayData(event, referralCode.id)
-            let referralCodeTraderDayData = getReferralCodeTraderDayData(referralCodeDayData.id, trader.id)
-
-            // uptick trading vol and fees for the referral code day tracking
-            referralCodeDayData.tradingVolume = referralCodeDayData.tradingVolume.plus(
-                abs(positionChangedEvent.exchangedPositionNotional),
-            )
-            referralCodeDayData.fees = referralCodeDayData.fees.plus(positionChangedEvent.fee)
-
-            // uptick trading vol and fees for the referral code day tracking for the trader
-            referralCodeTraderDayData.tradingVolume = referralCodeTraderDayData.tradingVolume.plus(
-                abs(positionChangedEvent.exchangedPositionNotional),
-            )
-            referralCodeTraderDayData.fees = referralCodeTraderDayData.fees.plus(positionChangedEvent.fee)
-
-            // uptick active traders for the referral code
-            let activeTraders = referralCodeDayData.activeReferees
-            if (!activeTraders.includes(trader.id)) {
-                activeTraders.push(trader.id)
-                referralCodeDayData.activeReferees = activeTraders
-            }
-
-            // Add the referrer code to the position changed event itself
-            positionChangedEvent.referralCode = code
-
-            referralCodeDayData.save()
-            referralCodeTraderDayData.save()
-            positionChangedEvent.save()
-        }
+    if (positionChangedEvent === null) {
+        return
+    }
+
+    const code = event.params.referralCode.toString()
+    const referralCode = getReferralCode(code)
+    if (referralCode === null) {
+        return
     }
+
+    const tradingVolume = abs(positionChangedEvent.exchangedPositionNotional)
+    const tradingFee = positionChangedEvent.fee
+
+    // uptick trading vol and fees for the referral code day tracking
+    const referralCodeDayData = getReferralCodeDayData(event, referralCode.id)
+    referralCodeDayData.tradingVolume = referralCodeDayData.tradingVolume.plus(tradingVolume)
+    referralCodeDayData.fees = referralCodeDayData.fees.plus(tradingFee)
+
+    // start from timestamp 1660435200 (2022-08-14T00:00:00.000Z)
```

Comment:
Just curious, why is 2022-08-14?

Reply:
We found wash trading activity for referral reward in the 3rd referral reward week (Aug 14-21), and also found that the subgraph handler `handleReferralPositionChanged ` has potential vulnerability which may be used to increase referral trader reward. So we adjusted the backend script first to apply new rule for 3rd week referral reward. Now we want to update the data on subgraph to reflect the same rule. 2022-08-14 the begin time of 3rd week.